### DISCUSSION TITLE
no workers satisfying: platform 'linux'

### DISCUSSION THREAD

**Discussion by martin-g** [concourse/concourse#6432] [Help & Support] [answered]
Hi!

I am trying to setup a worker running on an ARM64 machine.
For that I use https://github.com/neumayer/concourse-arm-worker. By executing its `build.sh` script on my ARM64 machine I get a Docker image for the worker.

The worker successfully connects to the web node (runs on AMD64)!

```
$ fly -t tutorial workers -d
name          containers  platform  tags     team  state    version  age     garden address   baggageclaim url        active tasks  resource types
arm64-worker  0           linux     aarch64  none  running  2.2      17m11s  127.0.0.1:45869  http://127.0.0.1:46845  0             registry-image
```

But when I try to run the following simple task:
```
---
platform: linux

image_resource:
  type: docker-image
  source: {repository: busybox}

run:
  path: echo
  args: [hello world]
```

 it fails with:
```
$ fly -t tutorial execute -c task_hello_world.yml
executing build 18 at http://127.0.0.1:8080/builds/18 
initializing
no workers satisfying: resource type 'docker-image', platform 'linux'
errored
```

I have the feeling that it is related to https://github.com/neumayer/concourse-arm-worker/tree/main/resource-types/registry-image but I am not sure.

I've replaced `docker-image` with `registry-image` but this didn't help.
I've also removed completely the `image_resource` but this just reduced the error to `no workers satisfying: platform 'linux'`

Any ideas ?

**Comment by martin-g** [comment, marked as answer]
Removing -e CONCOURSE_TAGS=aarch64 in the worker and changing `type: docker-image` with `type: registry-image` worked around the issue.

Now the problem is that it cannot use busybox ...

**Reply by taylorsilva** [reply]
To further clarify why you got the error, by default for resources, concourse will try and schedule them on an untagged linux worker and you only had a tagged linux worker connected to the cluster. That's why removing the tag resolved the issue for you.

Another way you could've worked around this is by adding the `aarch64` tag to your resources: https://concourse-ci.org/resources.html#schema.resource.tags

**Reply by martin-g** [reply]
@taylorsilva This does not work because this is a `task` that uses anonymous resource (`image_resource`) and according to https://concourse-ci.org/tasks.html#schema.anonymous_resource it does **not** support tags.

**Comment by muzhaqi16** [comment]
Replacing `type: docker-image` with ` type: registry-image` worked for me without any other changes.
